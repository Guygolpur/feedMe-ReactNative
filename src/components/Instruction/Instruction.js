import React, { Component } from 'react'
import { StyleSheet, Text, View, TouchableOpacity, ScrollView, ImageBackground } from 'react-native'
// import { WebView } from 'react-native-webview'
import { Icon } from 'react-native-elements'
import propTypes from 'prop-types'

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  titleContainer: {
    height: 80,
    flexDirection: 'row',
    paddingLeft: 30,
    width: '100%',
    backgroundColor: 'black',
    alignItems: 'center'
  },
  imagePos: {
    alignItems: 'center'
  },
  youtubeVideo: {
    height: 300,
    width: 300,
    paddingTop: 30,
  },
  backArrow: {
    paddingLeft: 30,
    left: 30
  },
  title: {
    fontSize: 35,
    fontWeight: 'bold',
    color: 'white',
    paddingLeft: 30
  },
  nameMeal: {
    fontSize: 25,
    textAlign: 'center',
    marginTop: 17
  },
  strArea: {
    textAlign: 'center',
    fontSize: 20
  },
  strInstructions: {
    textAlign: 'center',
    fontSize: 15,
    marginTop: 17
  }
})

export default class Instruction extends Component {
  constructor(props) {
    super(props)
    this.state = {
      instructions: {
        strMeal: '',
        strCategory: '',
        strArea: '',
        strInstructions: '',
        strMealThumb: '',
        strYoutube: ''
      },
      ready: false
    }
    this.getInstructionData = this.getInstructionData.bind(this)
    this.addInstruction = this.addInstruction.bind(this)
  }
  componentDidMount() {
    this.getInstructionData(this.props.recipe.name)
  }
  getInstructionData(MealName) {
    const url = `https://feedme24.herokuapp.com/getInstruction/${MealName}`
    fetch(`${url}`)
      .then(res => res.json())
      .then(data =>
        this.addInstruction(
          data[0].strMeal,
          data[0].strCategory,
          data[0].strArea,
          data[0].strInstructions,
          data[0].strMealThumb,
          data[0].strYoutube
        )
      )
      .catch(err => new Error(err))
  }
  addInstruction(
    strMeal = null,
    strCategory = null,
    strArea = null,
    strInstructions = null,
    strMealThumb = null,
    strYoutube = null
  ) {
    this.setState(
      {
        instructions: {
          strMeal,
          strCategory,
          strArea,
          strInstructions,
          strMealThumb,
          strYoutube
        }
      },
      () => {
        this.setState({ ready: true })
      }
    )
  }
  render() {
    return (
      <View style={styles.container}>
        <View style={styles.titleContainer}>
          <TouchableOpacity onPress={() => this.props.backToRecipeList()}>
            <Icon style={styles.backArrow} name="arrow-back" size={50} color={'white'} />
          </TouchableOpacity>
          <Text style={styles.title}>Instruction</Text>
        </View>
        <ScrollView>
          <Text style={styles.nameMeal}>{this.state.instructions.strMeal}</Text>
          <Text style={styles.strArea}>{this.state.instructions.strArea}</Text>
          <Text style={styles.strInstructions}>{this.state.instructions.strInstructions}</Text>
          {this.state.ready && (
            <View style={styles.imagePos}>
              {/* <WebView source={{ uri: `${this.state.instructions.strYoutube}` }} /> */}
              <ImageBackground style={styles.youtubeVideo} source={{ uri: this.state.instructions.strMealThumb }} />
            </View>
          )}
        </ScrollView>
      </View>
    )
  }
}
Instruction.propTypes = {
  recipe: propTypes.object,
  backToRecipeList: propTypes.func
}
