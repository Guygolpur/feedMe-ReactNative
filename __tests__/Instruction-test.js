import React from 'react'
import renderer from 'react-test-renderer'
import Instruction from '../src/components/Instruction/Instruction'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
configure({ adapter: new Adapter() })

if (!global.fetch) {
  global.fetch = require('node-fetch')
}
describe('Instruction component tests', () => {
  test('renders correctly', async () => {
    const mockFunck = await jest.fn()
    const tree = await renderer
      .create(<Instruction backToRecipeList={mockFunck} recipe={{ name: 'Budino Di Ricotta' }} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  test('check toHaveProperty props', async () => {
    const mockFunck = await jest.fn()
    const testRenderer = await renderer.create(
      <Instruction backToRecipeList={mockFunck} recipe={{ name: 'Budino Di Ricotta' }} />
    )
    expect(testRenderer.root.props).toHaveProperty('backToRecipeList')
    expect(testRenderer.root.props).toHaveProperty('recipe')
  })
  test('check addInstruction function', async () => {
    const mockFunck = await jest.fn()
    const wrapper = shallow(
      <Instruction backToRecipeList={mockFunck} recipe={{ name: 'Budino Di Ricotta' }} />
    )
    const componentInstance = wrapper.instance()
    expect(wrapper.state('ready')).toBe(false)
    expect(wrapper.state('instructions')).toStrictEqual({
      strMeal: '',
      strCategory: '',
      strArea: '',
      strInstructions: '',
      strMealThumb: '',
      strYoutube: ''
    })
    await componentInstance.addInstruction(
      'meal',
      'catagory',
      'strArea',
      'strInstructions',
      'strMealThumb',
      'strYoutube'
    )
    expect(wrapper.state('ready')).toBe(true)
    expect(wrapper.state('instructions')).toStrictEqual({
      strMeal: 'meal',
      strCategory: 'catagory',
      strArea: 'strArea',
      strInstructions: 'strInstructions',
      strMealThumb: 'strMealThumb',
      strYoutube: 'strYoutube'
    })
  })
  test('check calling backToRecipeList', async () => {
    const mockFunck = await jest.fn()
    const wrapper = await shallow(
      <Instruction backToRecipeList={mockFunck} recipe={{ name: 'Budino Di Ricotta' }} />
    ).instance()
    wrapper.props.backToRecipeList()
    expect(mockFunck).toHaveBeenCalled()
    expect(mockFunck).toHaveBeenCalledTimes(1)
  })
})
