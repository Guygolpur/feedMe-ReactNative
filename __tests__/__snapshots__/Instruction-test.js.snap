// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Instruction component tests renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "black",
        "flexDirection": "row",
        "height": 80,
        "paddingLeft": 30,
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            arrow-back
          </Text>
        </View>
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 35,
          "fontWeight": "bold",
          "paddingLeft": 30,
        }
      }
    >
      Instruction
    </Text>
  </View>
</View>
`;
